### DISCUSSION TITLE
About the unique of uuid

### DISCUSSION THREAD

**Discussion by fan-tastic-z** [loco-rs/loco#304] [General]
**Description**

After reading this https://github.com/loco-rs/loco/issues/245
I asked a question about whether it is best to use the pid field when establishing foreign keys and many-to-many relationships.
Through a simple code, it was discovered that if pid is used as a foreign key association, since uuid is unique by default in loco，As a result, multiple pieces of data cannot be added due to unique constraints.
```
/// Create a UUID column definition with a unique constraint.
pub fn uuid<T>(name: T) -> ColumnDef
where
    T: IntoIden,
{
    ColumnDef::new(name).unique_key().uuid().not_null().clone()
}
```
Therefore, I would like to ask whether you should consider letting users choose unique_key when using uuid?
In other words, when it comes to one-to-many and many-to-many, the default id field should still be used for association

**Comment by kaplanelad** [comment]
Move to discussions.

**Comment by fan-tastic-z** [comment]
I would like to ask if I am using this incorrectly? Should id be used when creating one-to-many and many-to-many?

**Comment by ronlobo** [comment]
Thoughts on CUID?

https://github.com/mplanchard/cuid-rust

**Reply by fan-tastic-z** [reply]
In fact, my main question is whether it is still necessary to use the default Id field when establishing one-to-many and many-to-many relationships in loco?
